pnpm-wireit/packages/beau/package.json

```diff
@@ -21,9 +21,26 @@
     }
   },
   "scripts": {
-    "build": "pnpm run build:esm && pnpm run build:cjs",
-    "build:esm": "babel src --out-dir es --config-file ../../.babelrc.js --copy-files --extensions .tsx,.ts,.jsx,.js",
-    "build:cjs": "BABEL_ENV=commonjs babel src --out-dir lib --config-file ../../.babelrc.js --copy-files --extensions .tsx,.ts,.jsx,.js"
+    "build": "wireit",
+    "build:esm": "wireit",
+    "build:cjs": "wireit"
+  },
+  "wireit": {
+    "build": {
+      "dependencies": ["build:esm", "build:cjs"]
+    },
+    "build:esm": {
+      "command": "nps pkg:build:esm",
+      "files": ["src/**", "../../.babelrc.js"],
+      "output": ["es/**"],
+      "dependencies": ["../core:build"]
+    },
```

Comment:
I'm bummed that `wireit` requires so much duplication.
I wanted to create a `clean` script, and had to add it to:
- in `root` `scripts > "clean": "pnpm --parallel run clean"`
- all `packages` in `scripts > "clean": "wireit"`
- all `packages` in `wireit > "clean": "nps pkg:clean"`
- in `root` `scripts > "pkg:clean": "rm -rf es lib"`

What a bummer that wireit doesn't have a way to inherit/extend a common config, and be DRY

Reply:
For `clean` specifically, Wireit should take care of wiping previous outputs.

... but get the general gist!

Reply:
Oh, and on the general duplication note, `nx` was much worse in this regard in terms of copy-pasta.

Wireit can't just take a root task and use it, but it _can_ take an arbitrary task, which I then use `nps` to be able to root define them. When you look at the boilerplate of what we're pasting here vs. what is actually going to change, I think it's acceptable in the sense that the inputs/outputs aren't likely going to change, the babel command (which _is_ abstracted) is...

Reply:
I pushed up my "clean" script, just because I wanted the exercise.  It didn't actually need `wireit`, so I got to skip a step.  Omitting wireit made me realize that the complexity in this configuration is due to our workspaces, not due to `wireit`.  Although I kinda wish `wireit` could've solved this problem for us :)

Reply:
@scottrippey Thanks for poking around and experimenting!

For the `clean` experiment, given that we use `nps` which _can_ be used to exec root commands from workspaces, I was able to simplify things in https://github.com/FormidableLabs/monorepo-infra-experiments/pull/6/commits/ff4c128991470a7bdcb1b23db2cab82bc7a96282 and remove all the `packages/*/package.json:scripts:clean` tasks and _just_ leave it in the root (switching `pnpm run` to `pnpm exec`)